src/index.ts

```diff
@@ -67,65 +63,84 @@ export function generateWords(nr: number, words: string[] = []): string {
     nr = Math.abs(nr);
   }
 
+  let remainder = 0;
+  let quotient: number;
+  let integer: number;
+  let decimals: number;
+
   switch (true) {
     case (nr < 20):
-      remainder = 0;
-      word = LESS_THAN_TWENTY[Math.trunc(nr)];
-      word += parseDecimals(nr);
+      [,, integer, decimals] = parseNumber(nr);
+      words.push(LESS_THAN_TWENTY[integer]);
+      if (decimals > 0) {
+        words.push(parseDecimals(decimals));
+      }
       break;
     case (nr < ONE_HUNDRED):
-      remainder = Math.trunc(nr % TEN);
-      word = TENTHS_LESS_THAN_HUNDRED[Math.floor(nr / TEN)];
+      [quotient, remainder] = parseNumber(nr, TEN);
+      words.push(TENTHS_LESS_THAN_HUNDRED[quotient]);
       // In case of remainder, we need to handle it here to be able to add the “ și ”
       if (remainder) {
-        word += ' și ';
+        words.push('și');
       }
       break;
     case (nr < ONE_THOUSAND):
-      remainder = nr % ONE_HUNDRED;
-      const hundreds = Math.floor(nr / ONE_HUNDRED);
-      switch (hundreds) {
-        case 1:
-          word = 'o sută';
-          break;
-        case 2:
-          word = 'două sute';
-          break;
-        default:
-          word = generateWords(hundreds) + ' sute';
-      }
+      [quotient, remainder] = parseNumber(nr, ONE_HUNDRED);
+      words.push(parseHundreds(quotient));
       break;
     case (nr < ONE_MILLION):
-      remainder = nr % ONE_THOUSAND;
-      const thousands = Math.floor(nr / ONE_THOUSAND);
-      word = match(thousands, 'o mie', 'mii');
+      [quotient, remainder] = parseNumber(nr, ONE_THOUSAND);
+      words.push(match(quotient, 'o mie', 'mii'));
```

Comment:
considering the current changes "match" is a very generic name compared to the above "parseHundreds", as they almost do the same thing, try to keep a naming consistency, as "match" is not enough to describe what this method actually does ore returns.